### PR TITLE
Add RandomRotation segmentation example (fill_mode constant vs crop)

### PR DESCRIPTION
**Pull request by PES2UG23CS205**
This PR adds a visual keras-io example demonstrating how
keras.layers.RandomRotation behaves for segmentation tasks.

It compares fill_mode="constant" vs fill_mode="crop" using
structured dict inputs to ensure synchronized transforms
for images and segmentation masks.

Related to keras-team/keras#21954

cc @innat @mattdangerw

### CODE REVIEW

**Comment by PES2UG23CS205** [comment]
Hi @innat 

I’ve opened a PR adding a keras-io example for RandomRotation focusing on
segmentation masks and comparing fill_mode="constant" vs "crop", with
visual before/after outputs.

PR: <https://github.com/keras-team/keras/pull/22011>

Thanks for the suggestion ,happy to iterate if you’d like any changes.

File: keras/src/layers/preprocessing/image_preprocessing/random_rotation.py [hunk 1]
```diff
@@ -75,6 +75,38 @@ class RandomRotation(BaseImagePreprocessingLayer):
             `image_data_format` value found in your Keras config file at
             `~/.keras/keras.json`. If you never set it, then it will be
             `"channels_last"`.
+    Examples:
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/random_rotation.py:78]: Oh, this part is already addressed here: https://github.com/keras-team/keras/pull/21978

File: examples/vision/random_rotation_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+"""
```
Comment (hertschuh) [inline examples/vision/random_rotation_segmentation.py:1]: We don't add examples in the keras repo like this because people won't find them and they won't see the pictures. The right place for discoverability is [keras.io](https://keras.io/) which you can contribute to here: https://github.com/keras-team/keras-io where images get rendered for guides and examples.

The thing is, we normally do end-to-end examples, having an example just about RandomRotation would be very narrow. However, we could really use a guide showing all the image augmentation layers.

Is that something you're willing to work on?

**Comment by PES2UG23CS205** [comment]
Hi @hertschuh,

Thanks for the guidance ,that makes sense.

Yes, I’m willing to work on a keras-io guide covering image augmentation layers.
I’ll close/move this example and start working on a proper keras-io guide instead.

Please let me know if there are any preferred structure or examples you'd like included.

**Comment by hertschuh** [comment]
@PES2UG23CS205 

Great, thanks!

I don't really have a preferred structure.
